storagemarket/impl/providerstates/provider_states.go

```diff
@@ -229,22 +229,7 @@ func WaitForPublish(ctx fsm.Context, environment ProviderDealEnvironment, deal s
 			return ctx.Trigger(storagemarket.ProviderEventDealPublishError, xerrors.Errorf("PublishStorageDeals error unmarshalling result: %w", err))
 		}
 
-		err = environment.SendSignedResponse(ctx.Context(), &network.Response{
-			State:          storagemarket.StorageDealProposalAccepted,
-			Proposal:       deal.ProposalCid,
-			PublishMessage: deal.PublishCid,
-		})
-
-		if err != nil {
-			return ctx.Trigger(storagemarket.ProviderEventSendResponseFailed, err)
-		}
-
-		if err := environment.Disconnect(deal.ProposalCid); err != nil {
```

Comment:
seems like disconnect code should move to RunDealDecider?
Also I believe you should take a look at ProviderEventDealPublished -- I believe that in that code it sets connectionClosed and you should probably move that to whatever the success Event on RunDealDecider.